kedro/io/core.py

```diff
@@ -281,11 +361,49 @@ def save(self: Self, data: _DI) -> None:
         return save
 
     def __init_subclass__(cls, **kwargs: Any) -> None:
-        """Decorate the `load` and `save` methods provided by the class.
+        """Customizes the behavior of subclasses of AbstractDataset during
+        their creation. This method is automatically invoked when a subclass
+        of AbstractDataset is defined.
+
 
+        Decorates the `load` and `save` methods provided by the class.
         If `_load` or `_save` are defined, alias them as a prerequisite.
 
         """
+
+        # Save the original __init__ method of the subclass
+        init_func: Callable = cls.__init__
+
+        def init_decorator(previous_init: Callable) -> Callable:
+            """A decorator that wraps the original __init__ of the subclass.
+
+            It ensures that after the original __init__ executes, the `__post_init__`
+            method of the instance is called with the arguments used to initialize
+            the object.
+            """
+
```

Comment:
```suggestion
@wraps(init_func)
```

and you need to import it at the top as `from functools import wraps`